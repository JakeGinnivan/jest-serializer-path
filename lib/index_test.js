'use strict'

const Serializer = require('./')

describe('serializer', () => {

  it('replaces __dirname with <PROJECT_ROOT>', () => {

    const sut = __dirname
    expect(sut).toMatchSnapshot()

  })


  it('preserves path directories after the <PROJECT_ROOT>', () => {

    const sut = `${__dirname}/src/somewhere`
    expect(sut).toMatchSnapshot()

  })


  it('replaces __dirname with <PROJECT_ROOT> in Object properties', () => {

    const sut = {
      myPath: `${__dirname}/src`,
    }
    expect(sut).toMatchSnapshot()

  })


  it('supports trailing slashes in the path', () => {

    const sut = `${__dirname}/path/with/trailing/slash/`
    expect(sut).toMatchSnapshot()

  })


  it('replaces every instance of __dirname in the same string', () => {

    const sut = {
      PATH: `${__dirname}/path:${__dirname}/another/path`,
      script: `const myPath = ${__dirname}/path;
      const mySecondPath = ${__dirname}/another/path;`,
    }
    expect(sut).toMatchSnapshot()

  })

})


describe('serializer.test()', () => {

  it('returns true when val is a string and contains __dirname', () => {

    const val = `${__dirname}/a/path`
    const result = Serializer.test(val)
    expect(result).toEqual(true)

  })

  it('returns false when val is a string and does not contain __dirname', () => {

    const val = '/a/path'
    const result = Serializer.test(val)
    expect(result).toEqual(false)

  })

  it('returns true when val is an object with a property that contains __dirname', () => {

    const val = {
      property: `${__dirname}/a/path`,
      property2: '/no/dirname',
    }
    const result = Serializer.test(val)
    expect(result).toEqual(true)

  })

  it('returns false when val is an object with a property that does not contain __dirname', () => {

    const val = {
      property: '/a/path/',
      property2: '/no/dirname',
    }
    const result = Serializer.test(val)
    expect(result).toEqual(false)

  })

})


describe('serializer.print()', () => {

  it('replaces __dirname with <PROJECT_ROOT> when val is a string', () => {

    const ser = jest.fn()
    const val = `${__dirname}/a/path`
    const expected = '<PROJECT_ROOT>/a/path'
    Serializer.print(val, ser)
    expect(ser.mock.calls[0][0]).toEqual(expected)

  })

  it('replaces __dirname with <PROJECT_ROOT> in string property when val is an object', () => {

    const ser = jest.fn()
    const val = {
      property: `${__dirname}/a/path`,
      property2: '/no/dirname',
    }
    const expected = {
      property: '<PROJECT_ROOT>/a/path',
      property2: '/no/dirname',
    }
    Serializer.print(val, ser)
    expect(ser.mock.calls[0][0]).toEqual(expected)

  })

})
