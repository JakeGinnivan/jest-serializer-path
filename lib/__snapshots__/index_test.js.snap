// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer preserves path directories after the <PROJECT_ROOT> 1`] = `"<PROJECT_ROOT>/src/somewhere"`;

exports[`serializer replaces __dirname with <PROJECT_ROOT> 1`] = `"<PROJECT_ROOT>"`;

exports[`serializer replaces __dirname with <PROJECT_ROOT> in Object properties 1`] = `
Object {
  "myPath": "<PROJECT_ROOT>/src",
}
`;

exports[`serializer replaces every instance of __dirname in the same string 1`] = `
Object {
  "PATH": "<PROJECT_ROOT>/path:<PROJECT_ROOT>/another/path",
  "script": "const myPath = <PROJECT_ROOT>/path;
      const mySecondPath = <PROJECT_ROOT>/another/path;",
}
`;

exports[`serializer supports trailing slashes in the path 1`] = `"<PROJECT_ROOT>/path/with/trailing/slash/"`;
